### PR TITLE
Fix elpa flags (missing optimization)

### PR DESCRIPTION
**Pull request by rasolca**
Add missing optimization flags.

### CODE REVIEW

**Review by albestro — DISMISSED** [review]

**Review by albestro — COMMENTED** [review]
@rasolca you explained me the reason and it makes sense.

can you add a comment about why it is needed to "force it" in this way?

**Comment by rasolca** [comment]
> @rasolca you explained me the reason and it makes sense.
> 
> can you add a comment about why it is needed to "force it" in this way?

Simply because ELPA configuration script doesn't add `-O3` automatically but it is left to the user to specify it.
See https://gitlab.mpcdf.mpg.de/elpa/elpa/-/blob/master/documentation/INSTALL.md for details.

**Comment by haampie** [comment]
Yeah, this is common in autotools... flags through variables don't compose but override defaults :(

Similar issue in #40387.

Passing flags through the compiler wrapper in `def flag_handler` is more composable, but has the downside that the effective flags do not appear in the build logs...

So, this is probably the best fix.

File: var/spack/repos/builtin/packages/elpa/package.py [hunk 1]
```diff
@@ -132,8 +132,8 @@ def configure_args(self):
             options.append("--enable-generic")
 
         if self.compiler.name == "gcc":
-            gcc_options = []
-            gfortran_options = ["-ffree-line-length-none"]
+            gcc_options = ["-O3"]
+            gfortran_options = ["-O3", "-ffree-line-length-none"]
```
Comment (haampie) [inline var/spack/repos/builtin/packages/elpa/package.py]: Feel free to simplify to `options.extend(["CFLAGS=-O3", "FFLAGS=-O3 -ffree-line-length-none"])` cause I don't see the point of `space_separator` :smile:

**Review by haampie — APPROVED** [review]

**Pull request merged**